include/seqan3/range/view/translation.hpp

```diff
@@ -33,256 +33,567 @@
 // ============================================================================
 
 /*!\file
+ * \ingroup view
  * \author Sara Hetzel <sara.hetzel AT fu-berlin.de>
- * \brief Provides seqan3::view::translate, seqan3::view::translate_single and seqan3::view::translate_frames.
+ * \brief Provides seqan3::view::translate and seqan3::view::translate_single.
  */
 
 #pragma once
 
+#include <iostream>
 #include <vector>
 #include <stdexcept>
 
 #include <seqan3/alphabet/nucleotide/dna5.hpp>
 #include <seqan3/alphabet/aminoacid/aa27.hpp>
 #include <seqan3/alphabet/aminoacid/translation.hpp>
 #include <seqan3/alphabet/aminoacid/translation_details.hpp>
+#include <seqan3/core/concept/core.hpp>
+#include <seqan3/core/add_enum_bitwise_operators.hpp>
 #include <seqan3/range/concept.hpp>
-#include <seqan3/range/view/complement.hpp>
+#include <seqan3/range/container/concept.hpp>
+#include <seqan3/range/view/detail.hpp>
+#include <seqan3/range/metafunction.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+
+namespace seqan3::detail
+{
+//!\brief Pop count function (calls gcc builtin function).
+template<integral_concept integral_type>
+auto pop_count(integral_type const n)
+{
+    return __builtin_popcount(n);
+}
 
-#include <range/v3/view/chunk.hpp>
-#include <range/v3/view/concat.hpp>
-#include <range/v3/view/drop.hpp>
-#include <range/v3/view/reverse.hpp>
-#include <range/v3/view/single.hpp>
-#include <range/v3/view/take_while.hpp>
-#include <range/v3/view/transform.hpp>
+}
 
 namespace seqan3
 {
-/*!\brief Specialization values for translation frames.
- * \details The numeric values of the enums correspond to the number of frames that are represented.
- */
-enum class translation_frames
+//!\brief Specialization values for single and multiple translation frames.
+enum class translation_frames : uint8_t
```

Comment:
please document the values of the enum, you can easily do this inline:
`FWD_FRAME_0  = 1,  //!\< FOOBAR`